Lesson15/Ayse/script.js

```diff
@@ -0,0 +1,504 @@
+isPalindrome('flower');
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+function evenOrOdd(number) {
+  if (number % 2 == 0) {
+    console.log(`${number}, is even.`);
+  } else {
+    console.log(`${number}, is odd.`);
+  }
+}
+evenOrOdd(5);
+evenOrOdd(10);
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+console.log("Example-11");
+
+function checkProtocol(url) {
+  const lowerCaseUrl = url.toLowerCase();
+  if (lowerCaseUrl.startsWith('https')) {
+    console.log('Secure connection');
+  } else {
+    console.log('Unsecure connection');
+  }
+}
+checkProtocol('https://www.google.com');
+checkProtocol('https://www.freecodecamp.org/');
+checkProtocol('//www.freecodecamp.org/');
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+console.log("Example-13");
+
+function repeatWord(word, times) {
+  const repeatWord = word.repeat(times);
+  console.log(repeatWord);
+}
+
+repeatWord("Hello", 3);
+repeatWord("Hi", 5);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+console.log("Example-14");
+
+function censorWord(sentence, target) {
+
+  const censoredSentence = sentence.replaceAll(target, '***');
+  console.log(censoredSentence);
+}
+
+censorWord('I love apples and apples are great.', 'apples');
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+console.log("Example-15");
+
+function startsWithA(str) {
+  if (str.charAt(0) == 'A') {
+    console.log('Starts with A');
+  } else {
+    console.log('Does not start with A');
+  }
+}
+startsWithA('Apple');
+startsWithA('Banana');
+
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+console.log("Example-16");
+
+function sliceLastN(text, n) {
+  const result = text.slice(-n);
+  console.log(result);
+}
+
+sliceLastN('Hello', 3);
+sliceLastN("JavaScript", 4);
+sliceLastN("Hello, World!", 1);
+
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+console.log("Example-17");
+
+function gradeChecker(score) {
+  switch (true) {
```

Comment:
This function is a good example of when using if-else produces much cleaner code that switch case. Prefer using if-else in cases like this in the future. No need to change anything now. 👍